termux-create-package.py

```diff
@@ -0,0 +1,128 @@
+        manifest_file_path = manifest
+        with open(manifest_file_path) as f: manifest = json.load(f)
+        for prop in 'name', 'version', 'files':
+            if prop not in manifest: sys.exit('Missing mandatory "' + prop + '" property')
+        package_name = manifest['name']
+        package_version = manifest['version'];
+        package_files = manifest['files']
+        package_arch = 'all'
+        if 'arch' in manifest: package_arch = manifest['arch']
+        if package_arch not in ['all', 'arm', 'i686', 'aarch64', 'x86_64']:
+            sys.exit('Invalid "arch" - must be one of all/arm/i686/aarch64/x86_64')
+        package_maintainer = 'None'
+        if 'maintainer' in manifest: package_maintainer = manifest['maintainer']
+        package_description = 'No description'
+        if 'description' in manifest: package_description = manifest['description']
+        package_homepage = 'No homepage'
+        if 'homepage' in manifest: package_homepage = manifest['homepage']
+        package_deps = []
+        if 'depends' in manifest: package_deps = manifest['depends']
+        package_provides = []
+        if 'provides' in manifest: package_provides = manifest['provides']
+        package_conflicts = []
+        if 'conflicts' in manifest: package_conflicts = manifest['conflicts']
+        output_debfile_name = package_name + '_' + package_version + '_' + package_arch + '.deb'
+        print('Building ' + output_debfile_name)
+        package_tmp_directory = tempfile.TemporaryDirectory()
+        with open(package_tmp_directory.name + '/debian-binary', 'w') as debian_binary: debian_binary.write("2.0\n")
+        with tarfile.open(package_tmp_directory.name + '/control.tar.xz', mode = 'w:xz') as control_tarfile:
+            contents = 'Package: ' + package_name + "\n"
+            contents += 'Version: ' + package_version + "\n"
+            contents += 'Architecture: ' + package_arch + "\n"
+            contents += 'Maintainer: ' + package_maintainer + "\n"
+            contents += 'Description: ' + package_description + "\n"
+            contents += 'Homepage: ' + package_homepage + "\n"
+            if len(package_deps) > 0:
+                contents += 'Depends: '
+                delim = ''
+                for d in package_deps:
+                    contents += delim + d
+                    delim = ','
+                contents += '\n'
+            if len(package_provides) > 0:
+                contents += 'Provides: '
+                delim = ''
+                for d in package_provides:
+                    contents += delim + d
+                    delim = ','
+                contents += '\n'
+            if len(package_conflicts) > 0:
+                contents += 'Conflicts: '
+                delim = ''
+                for d in package_conflicts:
+                    contents += delim + d
+                    delim = ','
+                contents += '\n'
```

Comment:
Maybe a for loop here with a small help function would be nice, since we do the same thing three times (I see it was the same in the old script as well)

Reply:
How d'you mean? I copied the source from @fornwall 's script and used what I could. I'm not too sure I understand some of it. I wanted to ask @fornwall anyway whether he could maybe add some comments to clarify what the code does in each section/line.

Reply:
I have tested it actually and it builds packages just fine on Mint and Ubuntu and Termux. However, one issue remains: When installing the built package using `APT`, it chucks an error into my face. I would like to know whether other people also experience this and if so, how to fix it.

Reply:
@theneonspectre could you post the error you get?

Reply:
Here you go:
```
Reading package lists... Error!
E: Sub-process Popen returned an error code (2)
E: Encountered a section with no Package: header
E: Problem with MergeList /home/myname/Desktop/TuxSDK/TuxSDK_2017.0_all.deb
E: The package lists or status file could not be parsed or opened.
```

This is the error I get when I try to run this command @Grimler91 
```
sudo apt install ./TuxSDK_2017.0_all.deb
```